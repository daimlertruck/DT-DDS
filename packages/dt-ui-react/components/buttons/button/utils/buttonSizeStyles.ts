import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
