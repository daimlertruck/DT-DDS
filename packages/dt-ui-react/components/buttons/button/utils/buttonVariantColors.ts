import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
