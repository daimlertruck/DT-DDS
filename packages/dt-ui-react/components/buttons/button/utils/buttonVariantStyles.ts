import { CustomTheme as Theme } from '../../../../types';
import { ButtonVariant, OutlinedColors, SolidColors } from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: SolidColors | OutlinedColors
) =>
  ({
    solid: `
      text-transform: uppercase;
      border: ${theme.spacing.none};

      &:focus {
        box-shadow: 0px 2px 1px 0px #1A1A1A0D, 0px 0px 0px 4px ${theme.colors.blue_10};
        background-color: ${theme.palette[color]};
      }

      &:disabled {
        opacity: 38%;
      }
  `,
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.neutralDark_100
            : theme.palette[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.neutralLight_100};
      }
    `,
  }[variant]);
