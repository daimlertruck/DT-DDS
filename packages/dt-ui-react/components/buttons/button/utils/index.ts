export * from './buttonVariantColors';
export * from './buttonVariantStyles';
