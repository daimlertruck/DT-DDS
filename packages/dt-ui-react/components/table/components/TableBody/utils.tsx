import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
