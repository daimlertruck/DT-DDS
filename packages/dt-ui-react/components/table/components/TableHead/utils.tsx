import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
