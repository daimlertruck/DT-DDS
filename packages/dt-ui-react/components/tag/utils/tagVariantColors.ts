import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
