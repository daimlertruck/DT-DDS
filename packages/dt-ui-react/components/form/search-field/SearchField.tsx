import { ChangeEvent } from 'react';

import { SearchIcon } from '../../../core';
import { debounce } from '../../../utils';
import { Spinner } from '../../spinner';
import { TextField } from '../text-field';

import { SearchFieldIconStyled } from './SearchField.styled';

export interface SearchFieldProps {
  label: string;
  initialValue?: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  onClick?: () => void;
  delay?: number;
  isLoading?: boolean;
}

export const SearchField = ({
  label,
  initialValue,
  onChange = () => null,
  onClick = () => null,
  delay = 1000,
  isLoading = false,
}: SearchFieldProps) => {
  const debouncedOnChange = debounce(onChange, delay);

  return (
    <TextField
      initialValue={initialValue}
      label={label}
      style={{
        width: '350px',
      }}
      onChange={debouncedOnChange}
    >
      <SearchFieldIconStyled
        {...(!isLoading && { onClick })}
        data-testid='search-field-icon'
        isLoading={isLoading}
      >
        {isLoading ? <Spinner size='small' /> : <SearchIcon />}
      </SearchFieldIconStyled>
    </TextField>
  );
};
