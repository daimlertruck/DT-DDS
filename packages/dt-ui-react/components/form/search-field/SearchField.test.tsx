import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../../utils';

import { SearchField } from './SearchField';

describe('<SearchField /> component', () => {
  jest.useFakeTimers();

  const ProvidedSearchField = withProviders(SearchField);

  it('renders SearchField input element with the correct label', () => {
    const { container } = render(
      <ProvidedSearchField label='Search for anything' />
    );

    expect(container).toMatchSnapshot();
  });

  it('fills Search input correctly with new value on change Event', () => {
    const { getByLabelText } = render(
      <ProvidedSearchField label='Search for anything' />
    );
    const input = getByLabelText('Search for anything');

    fireEvent.change(input, {
      target: { value: 'New search query value' },
    });

    expect(input).toHaveValue('New search query value');
  });

  it('calls the onChange callback when input changes', () => {
    const mockOnChange = jest.fn();

    const { container } = render(
      <ProvidedSearchField label='Search' onChange={mockOnChange} />
    );
    const input = container.querySelector('input') as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'test' } });
    jest.advanceTimersByTime(1000);

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  it('calls the onClick callback when the icon is clicked', () => {
    const mockOnClick = jest.fn();
    const { getByTestId } = render(
      <ProvidedSearchField label='Search' onClick={mockOnClick} />
    );

    const icon = getByTestId('search-field-icon');

    fireEvent.click(icon);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('renders SearchField input element with spinner when loading is true', () => {
    const { container } = render(
      <ProvidedSearchField label='Label' isLoading={true} />
    );

    expect(container).toMatchSnapshot();
  });
});
