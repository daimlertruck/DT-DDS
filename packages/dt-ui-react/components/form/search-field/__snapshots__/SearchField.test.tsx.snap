// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> component renders SearchField input element with the correct label 1`] = `
.emotion-0 {
  display: block;
  height: auto;
  width: 100%;
  position: relative;
}

.emotion-2 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #454545;
  cursor: inherit;
  left: 15px;
  top: 15px;
  position: absolute;
  -webkit-transform: translateY(0);
  -moz-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-4 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  background-color: #FFFFFF;
  border-radius: 0px;
  border: 1px solid #DEDEDE;
  color: #454545;
  min-width: 198px;
  padding: 20px 15px 5px 15px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  height: 53px;
}

.emotion-4:focus {
  outline: 1px solid #00677F;
  border: 1px solid transparent;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F3F3F5;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column nowrap;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  height: 35px;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  position: absolute;
  right: 10px;
  top: 10px;
  width: 35px;
}

.emotion-6 svg {
  fill: #454545;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.emotion-6:hover svg {
  fill: #292929;
}

<div>
  <div
    class="emotion-0 emotion-1"
    style="width: 350px;"
  >
    <label
      class="emotion-2 emotion-3"
      data-testid="label-field"
      for="Search-for-anything"
    >
      Search for anything
    </label>
    <input
      class="emotion-4 emotion-5"
      id="Search-for-anything"
      name="Search-for-anything"
      type="text"
      value=""
    />
    <span
      class="emotion-6 emotion-7"
    >
      <svg
        height="19"
        viewBox="0 0 18 19"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.8645 11.8208H12.0515L11.7633 11.5429C12.7719 10.3696 13.3791 8.84648 13.3791 7.18954C13.3791 3.49485 10.3842 0.5 6.68954 0.5C2.99485 0.5 0 3.49485 0 7.18954C0 10.8842 2.99485 13.8791 6.68954 13.8791C8.34648 13.8791 9.86964 13.2719 11.0429 12.2633L11.3208 12.5515V13.3645L16.4666 18.5L18 16.9666L12.8645 11.8208ZM6.68954 11.8208C4.12693 11.8208 2.05832 9.75214 2.05832 7.18954C2.05832 4.62693 4.12693 2.55832 6.68954 2.55832C9.25214 2.55832 11.3208 4.62693 11.3208 7.18954C11.3208 9.75214 9.25214 11.8208 6.68954 11.8208Z"
        />
      </svg>
    </span>
  </div>
</div>
`;
